packages/fx-core/src/plugins/resource/frontend/utils/environment-utils.ts

```diff
@@ -20,4 +22,50 @@ export class EnvironmentUtils {
     }
     await fs.writeFile(envFile, envs);
   }
+
+  static async updateEnvironment(
+    envFilePath: string,
+    runtimeEnv: RuntimeEnvironment,
+    aadEnv: AADEnvironment,
+    functionEnv?: FunctionEnvironment
+  ): Promise<void> {
+    let envs = await EnvironmentUtils.readEnvironments(envFilePath);
+
+    if (!envs) {
+      envs = {};
+    }
+
+    if (functionEnv) {
+      if (!envs[EnvironmentVariables.FuncName]) {
+        envs[EnvironmentVariables.FuncName] = functionEnv.defaultName;
```

Comment:
Do we need to add these env var even .env file exists?

Reply:
If .env file exists, we will add needed variable that not exist, to avoid runtime error for Tab App